mne/preprocessing/xdawn.py

```diff
@@ -0,0 +1,565 @@
+            self.signal_cov_ = _regularized_covariance(sig_data, self.reg)
+
+        # estimates evoked covariance
+        self.evokeds_cov_ = {}
+        if self.correct_overlap:
+            if epochs.baseline is not None:
+                raise ValueError('Baseline correction must be None if overlap '
+                                 'correction activated')
+            evo, to = least_square_evoked(epochs, return_toeplitz=True)
+        else:
+            evo = dict()
+            to = dict()
+            for eid in epochs.event_id:
+                evo[eid] = epochs[eid].average()
+                to[eid] = 1.0
+        self.evokeds_ = evo
+
+        for eid in epochs.event_id:
+            data = np.dot(evo[eid].data, to[eid])
+            self.evokeds_cov_[eid] = _regularized_covariance(data, self.reg)
+
+        # estimates spatial filters
+        for eid in epochs.event_id:
+
+            if self.signal_cov_.shape != self.evokeds_cov_[eid].shape:
+                raise ValueError('Size of signal cov must be the same as the'
+                                 ' number of channels in epochs')
+
+            evals, evecs = linalg.eigh(self.evokeds_cov_[eid],
+                                       self.signal_cov_)
+            evecs = evecs[:, np.argsort(evals)[::-1]]  # sort eigenvectors
+            evecs /= np.apply_along_axis(np.linalg.norm, 0, evecs)
+
+            self.filters_[eid] = evecs
+            self.patterns_[eid] = linalg.inv(evecs.T)
+
+            # Create and add projectors
+            ch_names = self.evokeds_[eid].ch_names
+            projs = []
+            for j in range(self.n_components):
+                proj_data = dict(col_names=ch_names, row_names=None,
+                                 data=evecs[:, j].T, nrow=1,
+                                 ncol=len(ch_names))
+                projs.append(Projection(active=True, data=proj_data,
+                             desc="%s Xdawn #%d" % (eid, j)))
+
+            self.projs_[eid] = projs
+
+        # store some values
+        self.ch_names = epochs.ch_names
+        self.exclude = range(self.n_components, len(self.ch_names))
+        self.event_id = epochs.event_id
+        return self
+
+    def transform(self, epochs):
+        """Apply Xdawn dim reduction.
+
+        Parameters
+        ----------
+        epochs : Epoch or array, shape=(n_trial x n_channels x n_times)
+            Data on which Xdawn filters will be applied.
+
+        Returns
+        -------
+        X : array, shape(n_trials x n_components * event_types x n_times)
+            Spatially filtered signals.
+        """
+        if isinstance(epochs, _BaseEpochs):
+            data = epochs.get_data()
+        elif isinstance(epochs, np.ndarray):
+            data = epochs
+        else:
+            raise ValueError('Data input must be of Epoch '
+                             'type or numpy array')
+
+        # create full matrix of spatial filter
+        V = []
```

Comment:
and if you can give it a name -- we try to avoid single character names outside of list comprehensions